### PR TITLE
[#151365021] API Definition Templates tests

### PR DESCRIPTION
**Pull request by gmereu**

### CODE REVIEW

File: lib/api/definitions/__tests__/MessageBodyMarkdown.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  test("isMessageBodyMarkdown on string shorter then 80 characters", () => {
+    // String shorter than 80 characters.
+    const text: string = "Lorem ipsum dolor sit amet";
+    expect(isMessageBodyMarkdown(text)).toBe(false);
+  });
+
+  test("isMessageBodyMarkdown on string longer then 10000 characters", () => {
+    // String long 90 characters.
+    const text: string =
```
Comment (gunzip) [inline lib/api/definitions/__tests__/MessageBodyMarkdown.test.ts]: maybe we should consider adding a libray like faker to generate fixtures
https://github.com/marak/Faker.js/

File: lib/api/definitions/__tests__/CreatedMessage.test.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+import { MessageContent } from "../MessageContent";
+
+import { toMessageSubject } from "../MessageSubject";
+
+import { toFiscalCode } from "../FiscalCode";
+
+import { toTimeToLive } from "../TimeToLive";
+
+describe("Check CreatedMessage methods", () => {
```
Comment (cloudify) [inline lib/api/definitions/__tests__/CreatedMessage.test.ts]: you should have a `describe` block for each method (you're testing each method)

then you should have a `test()` or `it()` block for each test you want to execute

for instance

```
describe("CreatedMessage#toCreatedMessage", () => {
  it("should return a defined option for valid payloads", () => {
  });

  it("should return an empty option for invalid payloads", () => {
  });
});
```

**Comment by cloudify** [comment]
@gmereu come procede? :)

**Comment by gmereu** [comment]
@cloudify ho revisionato la metà dei test. tra oggi e domani mattina dovrei avere terminato.

**Comment by gmereu** [comment]
@cloudify ho terminato la correzione dei test secondo le tue indicazioni.

File: lib/api/definitions/__tests__/CreatedMessage.test.ts [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+import { MessageContent } from "../MessageContent";
+
+import { toMessageSubject } from "../MessageSubject";
+
+import { toFiscalCode } from "../FiscalCode";
+
+import { toTimeToLive } from "../TimeToLive";
+
+describe("CreatedMessage#toCreatedMessage", () => {
+  test("should returns a defined option for valid payloads", () => {
```
Comment (cloudify) [inline lib/api/definitions/__tests__/CreatedMessage.test.ts]: you should use `it` instead of `test` like the rest of the tests

Comment (cloudify) [reply lib/api/definitions/__tests__/CreatedMessage.test.ts]: `returns` -> `return`

File: lib/api/definitions/__tests__/CreatedMessage.test.ts [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+    const messageOne = {
+      fiscal_code: toFiscalCode("AAABBB01C01A000A").get,
+      id: "12345",
+      sender_organization_id: "Sender Organization",
+      time_to_live: toTimeToLive(3600).get
+    };
+    expect(isCreatedMessage(messageOne)).toBe(true);
+  });
+  test("should returns true if CreatedMessage object does have content property set to null", () => {
+    /* tslint:disable */
```
Comment (cloudify) [inline lib/api/definitions/__tests__/CreatedMessage.test.ts]: do not disable tslint altogether, instead just disable the specific rules you need to disable

File: lib/api/definitions/__tests__/CreatedMessage.test.ts [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+    const message: CreatedMessage = {
+      content: messageContent,
+      fiscal_code: toFiscalCode("AAABBB01C01A000A").get,
+      id: "12345",
+      sender_organization_id: "Sender Organization",
+      time_to_live: toTimeToLive(3600).get
+    };
+
+    expect(toCreatedMessage(message).get).toEqual(message);
```
Comment (cloudify) [inline lib/api/definitions/__tests__/CreatedMessage.test.ts]: what's the point here? `message` the compiler already checks that `message` is a `CreatedMessage` - I think you should change your testing strategy here, what you want to do is to generate a number of fixtures for valid and invalid objects (either manually like [we do here](https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/models/__tests__/notification_event.test.ts#L43) or automatically as per @gunzip suggestion)

**Comment by gmereu** [comment]
@cloudify ho corretto i test seguendo il modello che i hai linkato.

**Comment by cloudify** [comment]
I'm closing this PR since the work is stalled

**Pull request closed without merging**